vendor/bat-native-ads/src/bat/ads/internal/conversions/conversions.cc

```diff
@@ -423,57 +423,52 @@ void Conversions::AddItemToQueue(
   });
 }
 
-bool Conversions::RemoveItemFromQueue(
-    const ConversionQueueItemInfo& conversion_queue_item) {
-  database::table::ConversionQueue database_table;
-  database_table.Delete(conversion_queue_item, [=](const bool success) {
-    if (!success) {
-      BLOG(0, "Failed to remove conversion from queue");
-      return;
-    }
-
-    BLOG(3, "Successfully removed conversion from queue");
-
-    StartTimerIfReady();
-  });
-
-  return true;
-}
-
 void Conversions::ProcessQueueItem(
     const ConversionQueueItemInfo& conversion_queue_item) {
+  const std::string& ad_type = conversion_queue_item.ad_type.ToString();
   const std::string& campaign_id = conversion_queue_item.campaign_id;
   const std::string& creative_set_id = conversion_queue_item.creative_set_id;
   const std::string& creative_instance_id =
       conversion_queue_item.creative_instance_id;
   const std::string& advertiser_id = conversion_queue_item.advertiser_id;
   const std::string& friendly_date_and_time =
-      LongFriendlyDateAndTime(conversion_queue_item.confirm_at);
+      LongFriendlyDateAndTime(conversion_queue_item.process_at);
 
   if (!conversion_queue_item.IsValid()) {
-    BLOG(1, "Failed to convert ad with campaign id "
-                << campaign_id << ", creative set id " << creative_set_id
+    BLOG(1, "Failed to convert "
+                << ad_type << " with campaign id " << campaign_id
+                << ", creative set id " << creative_set_id
                 << ", creative instance id " << creative_instance_id
                 << " and advertiser id " << advertiser_id << " "
                 << friendly_date_and_time);
 
     NotifyConversionFailed(conversion_queue_item);
```

Comment:
should we return here?

Reply:
it seems like `NotifyConversion(conversion_queue_item);` will be called for not valid items too

Reply:
Great catch as we should remove invalid item from queue